docs/tutorial/building-a-theme.md

```diff
@@ -0,0 +1,1504 @@
+
+### Configure the theme
+
+Open up the code for this new project.
+
+Create a new `gatsby-config.js` file in the root:
+
+```javascript:title=theme-test/gatsby-config.js
+module.exports = {
+  plugins: ["@jlengstorf/gatsby-theme-events"],
+}
+```
+
+### Install the Gatsby CLI
+
+Run:
+
+```shell
+yarn global add gatsby-cli
+```
+
+### Run the site
+
+Making sure you're in your `/theme-test` directory, run `gatsby develop` to start the site.
+
+![The new site, running your new Gatsby theme.](./images/building-a-theme-running-theme.png)
+
+### Add some data
+
+In your project, create a directory, `data`. Inside `data`, create a new file, `events.yml`:
+
+```yaml:title=theme-test/data/events.yml
+- name: Party
+  location: My House
+  start_date: 2019-06-26
+  end_date: 2019-06-26
+  url: https://jason.af/party
+```
+
+Save, and you'll see the new event data in your project:
+
+![The new site, with your new test event data.](./images/building-a-theme-add-test-event.png)
+
+## Use component shadowing to override theme components
+
+"Component shadowing" in Gatsby themes allow us to override or modify components in the theme. Let's use our new "theme-test" project to experiment with this.
+
+Start by adding a `src` folder to your project.
+
+Inside `src`, create a folder with the same title as your theme.
+
+> 💡 Note, with a namespaced theme, this will be two directories deep. For example, using the example of `@jlengstorf/gatsby-theme-events`, the file tree of your project will look like this:
+
+```text
+.
+├── data
+│   └── events.yml
+├── src
+│   └── @jlengstorf
+│       └── gatsby-theme-events
+├── gatsby-config.js
+├── package.json
+└── yarn.lock
+```
+
+Anything inside `theme-test/src/@jlengstorf/gatsby-theme-events` will "shadow" the `@jlengstorf/gatsby-theme-events` theme.
+
+### Override theme colors
+
+To test this shadowing, let's override `theme.js`:
+
+```javascript:title=theme-test/src/@jlengstorf/gatsby-theme-events/theme.js
+import { theme as baseTheme } from "@jlengstorf/gatsby-theme-events/src/theme"
+
+export const theme = {
+  ...baseTheme,
+  colors: {
+    ...baseTheme.colors,
+    primary: "blue",
+  },
+}
+```
+
+<!-- @TODO screenshot -->
+
+### Override an entire component
+
+To override an entire component, create a new file in your project, corresponding to that file in the theme.
+
+For example, create a new file in your "theme-test" project to override `layout.js`.
+
+```javascript:title=theme-test/src/@jlengstorf/gatsby-theme-events/components/layout.js
+import React from "react"
+
+export default ({ children }) => <>{children}</>
+```
+
+If you restart the development server, you'll see all of the styles and structure from the theme have been stripped away, because the component's been completely overridden.
+
+<!-- @TODO screenshot -->
```

Comment:
I'm sure you're already on this, but a recap would help to wrap up the tutorial. Adding a note here as a reminder!

Reply:
Yep, just added a conclusion!